run-tests

```diff
@@ -21,6 +27,11 @@ FLAGS=(--verbose --no-keep-going)
 # Also, add some compiler flags to verify standard compliance.
 export CFLAGS='-O0 -std=c99 -Wall -Werror -Wundef -Wpedantic -Wno-unused'
 
+# When testing the garbage collector, we use -O2 for a greater chance of finding bugs.
+# The HARDMEMTESTS macro makes lualib do garbage collection whenever it can.
+[ -n "$test_gc" ] &&
+    CFLAGS='-DHARDMEMTESTS -O2 -std=c99 -Wall -Werror -Wundef -Wpedantic -Wno-unused'
```

Comment:
Instead of copy-pasting the whole CFLAGS line, can we add just the -DHARDMEMTESTS to the rest of the line?

Is -O0 too slow? It would be easier to debug with gdb if it comes to that.

Reply:
I don't remember that. I think I might have suggested that out of the concern that it would make the tests run faster.

If -O0 is fast enough, let's stick with -O0